### PR TITLE
chore: added * pattern for test expectations

### PR DESCRIPTION
**Pull request by Lightning00Blade**
Add the ability to use `*` in test description where `*` is any `it` title.

### CODE REVIEW

File: tools/mochaRunner/src/utils.ts [hunk 1]
```diff
@@ -216,3 +219,28 @@ export function getTestId(file: string, fullTitle?: string): string {
     ? `[${getFilename(file)}] ${fullTitle}`
     : `[${getFilename(file)}]`;
 }
+
+export function testIdMatchesExpectationPattern(
```
Comment (OrKoN) [inline tools/mochaRunner/src/utils.ts:210]: Let's add a test in mocha runner tests for this?

File: tools/mochaRunner/src/utils.ts [hunk 2]
```diff
@@ -95,18 +95,17 @@ export function findEffectiveExpectationForTest(
...
-      return (
-        '' === expectation.testIdPattern ||
-        getTestId(result.file) === expectation.testIdPattern ||
-        getTestId(result.file, result.fullTitle) === expectation.testIdPattern
-      );
-    })
-    .pop();
+  return expectations.find(expectation => {
+    return (
+      '' === expectation.testIdPattern ||
```
Comment (OrKoN) [inline tools/mochaRunner/src/utils.ts]: do you want to require `*` and `[spec]*` instead of the current notation? It think it would be easier to maintain if we only support explicit *

File: tools/mochaRunner/src/main.ts [hunk 3]
```diff
@@ -200,17 +201,13 @@ async function main() {
...
-        );
-        if (recommendation.length > 0) {
+        const updates = getExpectationUpdates(results, applicableExpectations, {
+          platforms: [os.platform()],
+          parameters,
+        });
+        if (updates.length > 0) {
           fail = true;
-          recommendations.push(...recommendation);
+          recommendations = updates;
```
Comment (OrKoN) [inline tools/mochaRunner/src/main.ts]: I think the could be recommendations from previous test suites right?

File: tools/mochaRunner/src/interface.ts [hunk 4]
```diff
@@ -28,8 +28,6 @@ const skippedTests: Array<{testIdPattern: string; skip: true}> = process.env[
   ? JSON.parse(process.env['PUPPETEER_SKIPPED_TEST_CONFIG'])
   : [];
 
-skippedTests.reverse();
```
Comment (OrKoN) [inline tools/mochaRunner/src/interface.ts:31]: Why was this removed? the last matching pattern should apply, right?

Comment (Lightning00Blade) [reply tools/mochaRunner/src/interface.ts:31]: The TestExpectations are now reversed after we read them, that lets us simplify the search -> See [change].(https://github.com/puppeteer/puppeteer/pull/9870/files#diff-b7bd54671e35ed689bbd420e35dbd2b343b9595252279fa082f9e419a724c98bR109)

**Review by OrKoN — APPROVED** [review]
